src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -240,10 +260,10 @@ public int hashCode() {
     public static ModelMetadata fromString(String modelMetadataString) {
         String[] modelMetadataArray = modelMetadataString.split(DELIMITER, -1);
 
-        if (modelMetadataArray.length != 7) {
+        if (modelMetadataArray.length != 8 && modelMetadataArray.length != 7) {
```

Comment:
At the very least, I can definitely extract to an else/if and change the log message depending on the length.

Reply:
yes, I think you separate them out. Also, if you just put these checks into separate methods it will make it more clear in future what purpose of check is. For instance: `isMetadataCreatedOnOrAfter212(modelMetadataArray)` then it makes the code more readable.

Reply:
Sure, I can add that

Reply:
After thinking about this further, the main issue is that with a restart upgrade, even on the most recent version the length could be 7 or 8.  I don't see a way to check based on a variable, so I think the best option is to rework the method to make the code more readable.  Thoughts?